### ISSUE TITLE
Issues with pycord

### ISSUE THREAD

**Issue by rwf93** [robertwayne/cogwatch#30]
Hello, 

pycord doesn't allow for asynchronous setup functions in cogs. Unless I am missing something I am not entirely sure how I could work around this as I presume internally the watcher requires the ``setup`` function in a cog to be asynchronous. 

Extra information:

main.py
```py
import os
import discord
from cogwatch import Watcher

bot = discord.Bot()

@bot.event
async def on_ready():
    print(f"We have logged in as {bot.user}")
    
    watcher = Watcher(bot, path="cogs", preload=True, debug=False)
    await watcher.start()

bot.run(os.getenv("BOT_TOKEN"))
```

cogs/misc.py:
```py
from discord.ext import commands

class Misc(commands.Cog):
    def __init__(self, bot):
        self.bot = bot
        self._last_member = None

    @commands.slash_command(name = "ping")
    async def ping(self, ctx):
        await ctx.respond("Pong!")

# fails here
async def setup(bot):
    await bot.add_cog(Misc(bot))
```

Note, the version I am using is 2.4.1. I am also debating on whether this issue should be specific to the pycord project and not this project, though, it does mention that this API supports pycord.

**Comment by robertwayne** [comment]
Looks like there was a change at some point in pycord. I see they are returning a list from `load_extension`. I'll have a fix for this pushed shortly.

Thanks for letting me know!

**Comment by robertwayne** [comment]
Alright, this should be fixed in v3.3.0 - pushed to PyPi now.

The library supports both async and sync functions under-the-hood, so you can just remove the `async/await` keywords from your command and it will figure out what to do with them.

**Comment by rwf93** [comment]
Thank you!
